### DISCUSSION TITLE
Identifier Rules Regex

### DISCUSSION THREAD

**Discussion by KR-15** [vdemichev/DiaNN#602] [Q&A]
Hi,

I would like to know what are the different parsing rules that allow to put the gene name, protein.IDs, Protein.Name in the pg.matrix report ? When I talk about parsing rules here is to know the regex that allows to extract information of the FASTA header in the correct columns of the different outputs.

For instance, in MQ we can see those information here:

![image](https://user-images.githubusercontent.com/72909968/215446108-52f08173-a14d-491c-9e99-564f186d7dde.png)

Moreover, I wanted to know if there is an easy in DIA-NN via for instance the "additional option" windows to change those regex in order that they fit our fasta header as it is possible with MQ for instance. I am asking that because I wanted to use a FASTA database from OpenProt (https://openprot.org/p/download) where the header could look like this (non exhaustive example):

>Q96HA4-4|TX=9606 OS=Human GN=C1orf159 TA=ENST00000379319.5,NM_017891.4,ENST00000421241.6 PA=Q96HA4-4,ENSP00000368623.1,ENSP00000400736.2,NP_060361.4

As you can see they are different from what we usually have with uniprot databases.

Thank you for your help,

Best,

KR

**Comment by vdemichev** [comment]
Hi KR,

Uniprot FASTA format is supported, SGD I think too, not sure about anything else. Need to format Uniprot style using some R or Python package, if you wish DIA-NN to extract protein/organism information correctly.

Best,
Vadim
